stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1795,6 +1805,152 @@ impl NakamotoChainState {
         }
     }
 
+    pub fn calculate_signer_slots(
```

Comment:
Thinking about this a bit more, I don't think this method needs to exist.  Instead, the function `StacksChainState::make_reward_set()` (stackslib/src/chainstate/stacks/boot/mod.rs) should be updated to assign each reward set entry a signing key.  Furthermore, you'd want to add the signing key to `RawRewardSetEntry` included the signing key, such that `get_reward_addresses_pox_4()` loaded it out of the stacking record along with every other field.  From there, once you have the reward set, you'd simply copy the signing keys into the `.signers` contract.

Reply:
Yes, I agree, and I implemented a version of this (before this PR opened, otherwise, I probably would have waited) in my branch while working on the bitvector stuff:

https://github.com/stacks-network/stacks-core/blob/feat/stacker-bitvec/stackslib/src/chainstate/stacks/boot/mod.rs#L612

See the note here:

https://github.com/stacks-network/stacks-core/blob/feat/stacker-bitvec/stackslib/src/chainstate/nakamoto/coordinator/mod.rs#L69

Reply:
`make_signer_set` is invoked by `make_reward_set`, and the result is included in the `RewardSet` struct (so that it can be stored in the SortitionDB and emitted to event observers).